ifc_objects.py

```diff
@@ -117,15 +117,15 @@ def __getstate__(self):
         return getattr(self, "Type", None)
 
     def __setstate__(self, state):
-        if state and hasattr(state, "Type"):
-            self.Type = state
-        return None
+        self.loads(state)
 
     def dumps(self):
-        return
+        return getattr(self, "Type", None)
 
     def loads(self, _state):
-        return
+        if state and hasattr(state, "Type"):
+            self.Type = state
```

Comment:
this should be self.Type = _state, right?

```suggestion
            self.Type = _state
```